src/renderer/hooks/useNodeMenu.tsx

```diff
@@ -33,8 +35,24 @@ export const useNodeMenu = (
         useContext(GlobalContext);
     const { isDirectlyDisabled, canDisable, toggleDirectlyDisabled } = useDisabled;
 
+    const { getNode, getNodes, getEdges } = useReactFlow<NodeData, EdgeData>();
+
     return useContextMenu(() => (
         <MenuList className="nodrag">
+            <MenuItem
+                icon={<CopyIcon />}
+                onClick={() => {
+                    const node = getNode(id);
+                    if (node && !node.selected) {
+                        const nodeCopy = { ...node, selected: true };
+                        copyToClipboard([nodeCopy], []);
+                    } else {
+                        copyToClipboard(getNodes(), getEdges());
+                    }
```

Comment:
What is this? The `node &&` part of the condition is always true (otherwise, what are we right-clicking on?). And the rest of the condition... If the node is not selected, copy the node, otherwise copy the whole chain. How does that make sense? Right-clicking does not select a node, so just right-clicking + Copy any node will copy the whole chain.

Reply:
> The node && part of the condition is always true

False. The type from react flow for this helper is it returns a node or undefined. After all, I could be passing it a garbage id for all it cares. 

> And the rest of the condition... If the node is not selected, copy the node, otherwise copy the whole chain

I knew I should have left a comment explaining this... The copyToClipboard() function gets every selected node and connected edge between selected nodes. Under normal copying behavior, you are selecting nodes and ctrl+c'ing, so we pass in the entire chain in order to figure this out. 

When you just right click on a node to copy it, it's not necessarily selected, so what I do is if the node you're right clicking on isn't selected, then that means you just want to copy that one node. If it is selected, then other ones might be selected too, and therefore we'll pass the entire chain to it like normal ctrl+c behavior.

This makes it work exactly how you would expect it to in practice. Yeah this code is a bit weird but that's purely due to how that copy to clipboard function was implemented.